### ISSUE TITLE
interpret_parameters gives different values to factor parameters

### ISSUE THREAD

**Issue by mattansb** [easystats/effectsize#104]
Paging Dr. @DominiqueMakowski ...

### Same
``` r
library(effectsize)

fit1 <- lm(mpg ~ cyl + disp + wt + am, mtcars)
interpret_parameters(fit1)
#>     Parameter Effect_Size Interpretation
#> 1 (Intercept)          NA     very large
#> 2         cyl -0.52869107     very large
#> 3        disp  0.15225309          small
#> 4          wt -0.58175868     very large
#> 5          am  0.01068576           tiny
standardize_parameters(fit1)
#> Parameter   | Coefficient (std.) |         95% CI
#> -------------------------------------------------
#> (Intercept) |           6.87e-17 | [-0.16,  0.16]
#> cyl         |              -0.53 | [-0.90, -0.15]
#> disp        |               0.15 | [-0.36,  0.66]
#> wt          |              -0.58 | [-0.98, -0.19]
#> am          |               0.01 | [-0.21,  0.24]
```

### Different
```R
fit2 <- lm(mpg ~ factor(cyl) + disp + wt + am, mtcars)
interpret_parameters(fit2)
#>      Parameter Effect_Size Interpretation
#> 1  (Intercept)          NA     very large
#> 2 factor(cyl)6 -0.33632390          large
#> 3 factor(cyl)8 -0.46426414     very large
#> 4         disp  0.03356390           tiny
#> 5           wt -0.52749424     very large
#> 6           am  0.01169141           tiny
standardize_parameters(fit2)
#> Parameter    | Coefficient (std.) |         95% CI
#> --------------------------------------------------
#> (Intercept)  |               0.61 | [ 0.12,  1.11]
#> factor(cyl)6 |              -0.71 | [-1.22, -0.21]
#> factor(cyl)8 |              -1.05 | [-1.95, -0.15]
#> disp         |               0.03 | [-0.55,  0.62]
#> wt           |              -0.53 | [-0.94, -0.11]
#> am           |               0.01 | [-0.21,  0.24]
```

Same problem:
```R
mtcars$cyl_f <- factor(mtcars$cyl)
fit3 <- lm(mpg ~ cyl_f + disp + wt + am, mtcars)
interpret_parameters(fit3)
#>     Parameter Effect_Size Interpretation
#> 1 (Intercept)          NA     very large
#> 2      cyl_f6 -0.33632390          large
#> 3      cyl_f8 -0.46426414     very large
#> 4        disp  0.03356390           tiny
#> 5          wt -0.52749424     very large
#> 6          am  0.01169141           tiny
standardize_parameters(fit3)
#> Parameter   | Coefficient (std.) |         95% CI
#> -------------------------------------------------
#> (Intercept) |               0.61 | [ 0.12,  1.11]
#> cyl_f6      |              -0.71 | [-1.22, -0.21]
#> cyl_f8      |              -1.05 | [-1.95, -0.15]
#> disp        |               0.03 | [-0.55,  0.62]
#> wt          |              -0.53 | [-0.94, -0.11]
#> am          |               0.01 | [-0.21,  0.24]
```

<sup>Created on 2020-07-16 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by DominiqueMakowski** [comment]
interesting. do you think the issue is with interepret_ or with standardize_? 😬

**Comment by mattansb** [comment]
This seems to be due to factor effects being converted to cohen's d, with `convert_d_to_r()`, inside `interpret_parameters()`.

however, the values are not r, they are beta (standardized slopes) and so this conversion does not hold mathematically (also this behavior isn't documented - got an email about this)....

So not sure what the code is trying to do, but it isn't succeeding :/

``` r
library(effectsize)

fit <- lm(mpg ~ factor(am), mtcars)


standardize_parameters(fit)
#> Parameter   | Coefficient (std.) |         95% CI
#> -------------------------------------------------
#> (Intercept) |              -0.49 | [-0.87, -0.11]
#> factor(am)1 |               1.20 | [ 0.60,  1.80]

interpret_parameters(fit)
#>     Parameter Effect_Size Interpretation
#> 1 (Intercept)          NA     very large
#> 2 factor(am)1   0.5151545     very large

cohens_d(mpg ~ am, data = mtcars)
#> Cohen's d |         95% CI
#> --------------------------
#>     -1.48 | [-2.27, -0.67]
```

<sup>Created on 2020-07-16 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by DominiqueMakowski** [comment]
yes I think the goal was to convert everything to r to use interpreter, because otherwise we need to detect wether to use interpret_r for slopes and interpret_d for factors if I'm not mistaken

**Comment by mattansb** [comment]
If that's what the function is suppose to due, I'm sure I understand the internal workings 😅

What does `.standardize_standardized_interactions()` do?

Why isn't the function "just":
```R
.interpret_parameters_regressions <- function(model, parameters = NULL, interpretation = "funder2019", standardize_method = "refit", standardize_robust = FALSE, ...) {
  std_es <- standardize_parameters(model, method = standardize_method, robust = standardize_robust, ...)
  
  data.frame(
    Parameter = std_es$Parameter,
    Effect_Size = std_es$Std_Coefficient,
    Interpretation = interpret_r(std_es$Std_Coefficient, rules = interpretation)
  )
}
```

Could you have a go at this?

**Comment by DominiqueMakowski** [comment]
Essentially, I think the problem that I was trying to solve (but did not fully succeed to my knowledge) is to correctly estimate standardized interaction without refitting the model.

Assuming the gold standard were the parameters from a model fitted on standardized data, I was trying to recover these parameters a posteriori, using the non-standardized parameters and the data. So for simpler models it works well, you can simply scale by the standard deviations of the outcome and the predictor. But for complex interactions this gets more tricky

I will look into it asap

**Comment by mattansb** [comment]
@DominiqueMakowski should I wait on this bugfix before releasing 0.3.2?

**Comment by DominiqueMakowski** [comment]
nah I'm swamped 🙉 Maybe we can add a warning saying that it's broken for now or something

**Comment by strengejacke** [comment]
He's doing some lame python stuff...

**Comment by DominiqueMakowski** [comment]
even worse, I have to do some actual writing 😢

**Comment by strengejacke** [comment]
I _promise_ I continue working on the prior paper this week...

**Comment by DominiqueMakowski** [comment]
![image](https://user-images.githubusercontent.com/8875533/88541495-6d1b9100-d047-11ea-85bb-4cce00489f2c.png)

**Comment by mattansb** [comment]
@DominiqueMakowski *bump*...

**Comment by mattansb** [comment]
Let me know when I get too aggressive (:
``` r
fit <- lm(mpg ~ factor(cyl) + disp + wt + am, mtcars)
effectsize::interpret_parameters(fit)
#> Warning: There is a known bug in 'interpret_parameters()' when interpreting parameters related to factors.
#> See github.com/easystats/effectsize/issues/104
#> This bug has been open for 62 days... :(
#>      Parameter Effect_Size Interpretation
#> 1  (Intercept)          NA     very large
#> 2 factor(cyl)6 -0.33632390          large
#> 3 factor(cyl)8 -0.46426414     very large
#> 4         disp  0.03356390           tiny
#> 5           wt -0.52749424     very large
#> 6           am  0.01169141           tiny
```

<sup>Created on 2020-09-16 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by mattansb** [comment]
@DominiqueMakowski If you don't mind me undoing all of your interaction interpretations / calculations,
can I just replace with:

```R
interpret_parameters.lm<- function(model, parameters = NULL, interpretation = "funder2019", standardize_method = "refit", standardize_robust = FALSE, ...) {
  std_es <- standardize_parameters(model, method = standardize_method, robust = standardize_robust, ...)
  
  data.frame(
    Parameter = std_es$Parameter,
    Effect_Size = std_es$Std_Coefficient,
    Interpretation = interpret_r(std_es$Std_Coefficient, rules = interpretation)
  )
}
```

**Comment by DominiqueMakowski** [comment]
okay okay if that puts your mind at ease, I'll open a new issue to discuss first theoretically what I was trying to achieve here

**Comment by mattansb** [comment]
Closing now, as after reading #127 I no longer believe this to be a bug - it is, in fact, a feature! (But one that needs more work...)
